examples/centralized_threshold.rs

```diff
@@ -0,0 +1,372 @@
+
+    println!(" - Soundness parameter: {soundness_param}");
+    println!(" - Completeness parameter: {completeness_param}");
+    println!(" - Prover set size: {set_size}");
+    println!(" - Lower bound: {lower_bound}");
+
+    // Initialize signers
+    println!("--------------------------------------------------------");
+    let mut signers: Vec<Signer> = (0..nb_elements as usize)
+        .map(|_| Signer::init(&mut rng))
+        .collect();
+    println!("-- {} signers initialized.", signers.len());
+
+    // Start new registration process
+    let mut registration = Registration::new();
+    println!("-- Registration is opened.");
+
+    // Register signer candidates
+    let mut count: u64 = 0;
+    let register_range = rng.gen_range(set_size..nb_elements);
+    for signer in signers.iter_mut().take(register_range as usize) {
+        if signer.register(&mut registration) {
+            count = count.saturating_add(1);
+        }
+    }
+    println!("-- {count} signers are registered.");
+
+    // Close the registration process
+    registration.close::<DATA_LENGTH>();
+    println!("-- Registration is closed.");
+
+    for signer in signers.iter_mut().take(register_range as usize) {
+        signer.get_closed_registration(&registration);
+    }
+
+    // Create individual signatures
+    let signature_range = rng.gen_range(set_size..register_range);
+    let signature_list: Vec<IndividualSignature> = signers
+        .iter()
+        .take(signature_range as usize)
+        .filter_map(|signer| signer.sign::<DATA_LENGTH>(msg))
+        .collect();
+    println!("-- {} signatures generated.", signature_list.len());
+
+    println!("--------------------------------------------------------");
+    println!("--------- Generating Alba threshold signature. ---------");
+
+    // Generate AlbaThresholdSignature proof
+    if let Some(alba_threshold_signature) =
+        AlbaThresholdSignature::prove::<DATA_LENGTH>(&alba, &signature_list, &registration, msg)
+    {
+        println!("-- Alba threshold signature is generated.");
+        println!("--------------------------------------------------------");
+        println!("--------- Verifying Alba threshold signature. ----------");
+
+        // Verify the proof
+        if alba_threshold_signature.verify::<DATA_LENGTH>(&alba, &registration, msg) {
+            println!("-- Verification successful.");
+        } else {
+            println!("-- Verification failed.");
+        }
+        println!("--------------------------------------------------------");
+    } else {
+        println!("-- Failed to generate Alba threshold signature.");
+    }
+}
+
+// Some basic tests
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use rand_core::RngCore;
+
+    // Error: Registration is not closed. Cannot verify signatures.
+    // Number of collected valid signatures: 0.
+    #[test]
+    fn test_collect_valid_signatures_no_closed_reg() {
```

Comment:
Since this is a test, can you add an assert on the nb of valid sigs and print only if the test fails?